saleor/order/models.py

```diff
@@ -99,7 +100,13 @@ class Order(ModelWithMetadata):
     shipping_method_name = models.CharField(
         max_length=255, null=True, default=None, blank=True, editable=False
     )
-
+    channel = models.ForeignKey(
+        Channel,
+        related_name="orders",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
```

Comment:
We shouldn't allow orders without channels. When someone remove a channel all related orders should be moved to new target channel .

Reply:
Removing channel is a separate task.